### ISSUE TITLE
Bug: Unable to set input to 'writtenbook' (item id 387)

### ISSUE THREAD

**Issue by Link2006** [Civcraft/ItemExchange#22]
It seems that whenever an item rule is set to written books, wether by /ies c writtenbook or /ies m 387, makes itemexchange no longer accept the item rule at all.

This makes donation boxes that would accept signed books unable to accept said books due to this problem.

Steps to replicate:
1) Create an input rule (Any item works)
2) Set it to written books using /ies c writtenbook
3a) Try using the chest with the rule (Fails)
3b) Try to change the rule

**Comment by ProgrammerDan** [comment]
Thanks, I'll look into it.

**Comment by Maxopoly** [comment]
It's because of the special bookmeta, afaik it has always been like that. Same for fireworks, dyed leather armour, maps etc. 

The way the exchange rules are coded isn't really the best and it's all legacy code. We could probably at some point rewrite the whole thing with the ItemMap stuff from civmodcore to even allow wildcarding certain aspects in shops, but for now this is very low priority and something we can consider once 3.0 is stable.

**Comment by ProgrammerDan** [comment]
Yeah I figured that might be the case. TBH this has been on my list for a while, mostly because I want "brewery" potions to be sale-able. You can do it currently via a redstone mechanism, but not directly. 

For donation boxes iirc the only option might be an "insecure" hopper over a chest.

**Comment by Maxopoly** [comment]
Hmm, especially with the item name api in civmodcore this would be doable in a really nice way, but I think at this point the best approach to this plugin is a complete rewrite if we want to add features like that. Even trying to update the legacy ChatColors completly broke the plugin last time.

**Comment by ProgrammerDan** [comment]
Yeah. I'm not a huge fan of rewriting but considering this plugin has fewer externalities (no db, etc.) it should be straightforward to the extreme.

**Comment by Maxopoly** [comment]
Yeah, also it's not too complex.
